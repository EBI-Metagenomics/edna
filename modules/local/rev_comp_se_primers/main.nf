
process REV_COMP_SE_PRIMERS {
    tag "$meta.id"
    label 'very_light'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:${params.mpt_version}":
        "biocontainers/mgnify-pipelines-toolkit:${params.mpt_version}" }"

    input:
    tuple val(meta), path(final_concat_primers)

    output:
    tuple val(meta), path("*rev_comp_se_primers.fasta"), optional: true, emit: rev_comp_se_primers_out
    path "versions.yml"                                , emit: versions

    script:
    """
    if [[ -s $final_concat_primers && ${meta.single_end} = true ]]; then
        rev_comp_se_primers -i $final_concat_primers -s ${meta.id} -o ./
    elif [[ -s $final_concat_primers && ${meta.single_end} = false ]]; then
        cat $final_concat_primers > ./${meta.id}_rev_comp_se_primers.fasta
    else
        touch ./${meta.id}_rev_comp_se_primers.fasta
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: ${params.mpt_version}
    END_VERSIONS
    """

}
