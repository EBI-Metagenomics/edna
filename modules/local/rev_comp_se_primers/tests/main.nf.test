nextflow_process {

    name "Test Process REV_COMP_SE_PRIMERS"
    script "../main.nf"
    process "REV_COMP_SE_PRIMERS"
    tag "modules"
    tag "rev_comp_se_primers"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('${projectDir}/modules/local/rev_comp_se_primers/tests/data/test_final_concat_primers.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.rev_comp_se_primers_out).match("rev_comp_se_primers_out") }
            )
        }

    }

}
