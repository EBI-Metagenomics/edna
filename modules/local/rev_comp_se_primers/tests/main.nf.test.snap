{
    "rev_comp_se_primers_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test_rev_comp_se_primers.fasta:md5,f26cdc22a5231d2dbd0831f1f1983dc6"
                ]
            ]
        ],
        "timestamp": "2023-11-06T16:07:34.765990196"
    }
}